src/main/java/org/folio/circulation/resources/RequestFromRepresentationService.java

```diff
@@ -116,10 +119,21 @@
   }
 
   private CompletableFuture<Result<Request>> fetchItemAndLoan(Request request) {
-    return succeeded(request)
-      .combineAfter(itemRepository::fetchFor, Request::withItem)
-      .thenComposeAsync(r -> r.after(this::fetchLoan))
-      .thenComposeAsync(r -> r.combineAfter(this::getUserForExistingLoan, this::addUserToLoan));
+      return succeeded(request)
+        .after(this::fetchItem)
+        .thenComposeAsync(r -> r.after(this::fetchLoan))
+        .thenComposeAsync(r -> r.combineAfter(this::getUserForExistingLoan, this::addUserToLoan));
+  }
+
+  private CompletableFuture<Result<Request>> fetchItem(Request request) {
+    if (request.getTlrSettingsConfiguration().isTitleLevelRequestsFeatureEnabled()
+      && request.getRequestType() == RequestType.PAGE) {
+      return itemByInstanceIdFinder.getFirstAvailableItemByInstanceId(request.getInstanceId())
+        .thenApply(r -> r.map(request::withItem));
+    } else {
+      return itemRepository.fetchFor(request)
+        .thenApply(r -> r.map(request::withItem));
+    }
```

Comment:
This method is called not only for newly created requests, if I remember correctly. Is there a case when it's called for an existing Page request with itemId already set? In this case we wouldn't need to look for another available item because we already have one and it's linked to the request.

Reply:
Done